docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,987 @@
+                callback(.failure(error))
+            }
+        }
+    }
+    
+    private var recorderDelegate: RecorderDelegate?
+    private var recorder: AVAudioRecorder?
+    
+    private class RecorderDelegate: NSObject, AVAudioRecorderDelegate {
+        private let callback: RecordingDoneCallback
+        
+        init(callback: @escaping RecordingDoneCallback) {
+            self.callback = callback
+        }
+        
+        func audioRecorderDidFinishRecording(
+            _ recorder: AVAudioRecorder,
+            successfully flag: Bool
+        ) {
+            let recordResult = RecordResult { () -> Data in
+                guard flag else {
+                    throw AudioRecorderError.Error(message: "Recording was unsuccessful.")
+                }
+                
+                let recordingUrl = recorder.url
+                let recordingFile = try AVAudioFile(forReading: recordingUrl)
+                
+                guard
+                    let format = AVAudioFormat(
+                        commonFormat: .pcmFormatFloat32,
+                        sampleRate: recordingFile.fileFormat.sampleRate,
+                        channels: 1,
+                        interleaved: false)
+                else {
+                    throw AudioRecorderError.Error(message: "Failed to create audio format.")
+                }
+                
+                guard
+                    let recordingBuffer = AVAudioPCMBuffer(
+                        pcmFormat: format,
+                        frameCapacity: AVAudioFrameCount(recordingFile.length))
+                else {
+                    throw AudioRecorderError.Error(message: "Failed to create audio buffer.")
+                }
+                
+                try recordingFile.read(into: recordingBuffer)
+                
+                guard let recordingFloatChannelData = recordingBuffer.floatChannelData else {
+                    throw AudioRecorderError.Error(message: "Failed to get float channel data.")
+                }
+                
+                return Data(bytes: recordingFloatChannelData[0], count: Int(recordingBuffer.frameLength) * MemoryLayout<Float>.size)
+               
+            }
+            
+            callback(recordResult)
+        }
+        
+        func audioRecorderEncodeErrorDidOccur(
+            _ recorder: AVAudioRecorder,
+            error: Error?
+        ) {
+            if let error = error {
+                callback(.failure(error))
+            } else {
+                callback(.failure(AudioRecorderError.Error(message: "Encoding was unsuccessful.")))
+            }
+        }
+    }
+}
+```
+
+### Training View
+
+The `TrainingView` will be used to train the model on the user's voice. First, it wil prompt the user to record `kNumRecordings` of their voice. Then, it will train the model on the user's voice and some pre-recorded audio data. Finally, it will export the trained model for inference purposes.
```

Comment:
```suggestion
The `TrainingView` will be used to train the model on the user's voice. First, it will prompt the user to record `kNumRecordings` recordings of their voice. Then, it will train the model on the user's voice and some pre-recorded recordings of other speakers' voices. Finally, it will export the trained model for inference purposes.
```